pkg/cmd/cli/cli_changefeed_create.go

```diff
@@ -0,0 +1,495 @@
+		o.commonChangefeedOptions.sortEngine = model.SortInMemory
+		log.Warn("The TiCDC cluster is built from an older version, disabling Unified Sorter by default",
+			zap.String("version", cdcClusterVer.String()))
+	}
+
+	if len(o.commonChangefeedOptions.configFile) > 0 {
+		if err := o.commonChangefeedOptions.strictDecodeConfig("TiCDC changefeed", cfg); err != nil {
+			return nil, err
+		}
+	}
+
+	if o.disableGCSafePointCheck {
+		cfg.CheckGCSafePoint = false
+	}
+
+	if o.commonChangefeedOptions.cyclicReplicaID != 0 || len(o.commonChangefeedOptions.cyclicFilterReplicaIDs) != 0 {
+		if !(o.commonChangefeedOptions.cyclicReplicaID != 0 && len(o.commonChangefeedOptions.cyclicFilterReplicaIDs) != 0) {
+			return nil, errors.New("invalid cyclic config, please make sure using " +
+				"nonzero replica ID and specify filter replica IDs")
+		}
+
+		filter := make([]uint64, 0, len(o.commonChangefeedOptions.cyclicFilterReplicaIDs))
+		for _, id := range o.commonChangefeedOptions.cyclicFilterReplicaIDs {
+			filter = append(filter, uint64(id))
+		}
+
+		cfg.Cyclic = &config.CyclicConfig{
+			Enable:          true,
+			ReplicaID:       o.commonChangefeedOptions.cyclicReplicaID,
+			FilterReplicaID: filter,
+			SyncDDL:         o.commonChangefeedOptions.cyclicSyncDDL,
+			// TODO(neil) enable ID bucket.
+		}
+	}
+
+	if !cfg.EnableOldValue {
+		sinkURIParsed, err := url.Parse(o.commonChangefeedOptions.sinkURI)
+		if err != nil {
+			return nil, cerror.WrapError(cerror.ErrSinkURIInvalid, err)
+		}
+
+		protocol := sinkURIParsed.Query().Get("protocol")
+		for _, fp := range forceEnableOldValueProtocols {
+			if protocol == fp {
+				log.Warn("Attempting to replicate without old value enabled. CDC will enable old value and continue.", zap.String("protocol", protocol))
+				cfg.EnableOldValue = true
+				break
+			}
+		}
+
+		if cfg.ForceReplicate {
+			log.Error("if use force replicate, old value feature must be enabled")
+			return nil, cerror.ErrOldValueNotEnabled.GenWithStackByArgs()
+		}
+	}
+
+	for _, rules := range cfg.Sink.DispatchRules {
+		switch strings.ToLower(rules.Dispatcher) {
+		case "rowid", "index-value":
+			if cfg.EnableOldValue {
+				cmd.Printf("[WARN] This index-value distribution mode "+
+					"does not guarantee row-level orderliness when "+
+					"switching on the old value, so please use caution! dispatch-rules: %#v", rules)
+			}
+		}
+	}
+
+	switch o.commonChangefeedOptions.sortEngine {
+	case model.SortUnified, model.SortInMemory:
+	case model.SortInFile:
+		// obsolete. But we keep silent here. We create a Unified Sorter when the owner/processor sees this option
+		// for backward-compatibility.
+	default:
+		return nil, errors.Errorf("Creating changefeed with an invalid sort engine(%s), "+
+			"`%s` and `%s` are the only valid options.", o.commonChangefeedOptions.sortEngine, model.SortUnified, model.SortInMemory)
+	}
+
+	info := &model.ChangeFeedInfo{
```

Comment:
Or we can create `info` after all necessary check done？

Reply:
See: https://github.com/pingcap/ticdc/pull/2539#issuecomment-900214332